src/main/java/svc/controllers/CourtController.java

```diff
@@ -30,11 +34,20 @@ CourtsDTO GetCourts() {
 	
 	@ResponseBody
 	@RequestMapping(method = RequestMethod.GET, value="/{id}")
-	Court GetCourt(@PathVariable("id") Long id) {
-		if (id == null) {
-			LogSystem.LogEvent("Null id passed to controller");
+	Court GetCourt(@PathVariable("id") Long id) throws NotFoundException {
+		Court court = courtManager.GetCourtById(id);
+		if (court == null) {
+			throw new NotFoundException("Court Not Found");
 		}
-		
-		return courtManager.GetCourtById(id);
+		return court;
 	}
+	
+	@ExceptionHandler(TypeMismatchException.class)
+	@ResponseStatus(value = HttpStatus.BAD_REQUEST, reason="Invalid Court ID")
+	public void typeMismatchExceptionHandler(TypeMismatchException e, HttpServletResponse response){	

```

Comment:
Does this indicate that we should move this to a base class, so all the controllers get it?